controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller.go

```diff
@@ -91,23 +95,22 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		}
 		toolchainPresenceReason = toolchainv1alpha1.SpaceProvisionerConfigToolchainClusterNotFoundReason
 		toolchainPresent = corev1.ConditionFalse
-	}
-
-	spaceProvisionerConfig.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(spaceProvisionerConfig.Status.Conditions,
-		toolchainv1alpha1.Condition{
-			Type:    toolchainv1alpha1.ConditionReady,
-			Status:  toolchainPresent,
-			Reason:  toolchainPresenceReason,
-			Message: toolchainPresenceMessage,
-		})
-
-	logger.Info("updating SpaceProvisionerConfig", "status", toolchainPresent, "reason", toolchainPresenceReason)
-	if err := r.Client.Status().Update(ctx, spaceProvisionerConfig); err != nil {
-		if reportedError != nil {
-			logger.Info("failed to update the status (reported as failed reconciliation) with a previous unreported error during reconciliation", "unreportedError", reportedError)
+	} else {
+		for _, c := range toolchainCluster.Status.Conditions {
+			if c.Type == toolchainv1alpha1.ToolchainClusterReady {
+				toolchainPresent = c.Status
+				break
+			}
+		}
+		if toolchainPresent != corev1.ConditionTrue {
```

Comment:
I believe we can simplify this to
```suggestion
		if condition.IsNotTrue(toolchainCluster.Status.Conditions, toolchainv1alpha1.ToolchainClusterReady) {
			toolchainPresent = apiv1.ConditionFalse
```

Reply:
These are cluster conditions which are different from the "ordinary" conditions that have the utility functions in the `condition` package. Note that there is currently PR open to get rid of the cluster conditions and replace them with the "ordinary" ones.

Reply:
Ah, missed that 👍